Composer/packages/client/src/pages/botProject/CreatePublishProfileDialog.tsx

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/** @jsx jsx */
+import React, { Fragment, useState, useEffect } from 'react';
+import { jsx } from '@emotion/core';
+import { useRecoilValue } from 'recoil';
+import { PublishTarget } from '@bfc/shared';
+import formatMessage from 'format-message';
+import { ActionButton, DefaultButton } from 'office-ui-fabric-react/lib/Button';
+
+import { dispatcherState, settingsState, publishTypesState } from '../../recoilModel';
+import { AuthDialog } from '../../components/Auth/AuthDialog';
+import { isShowAuthDialog } from '../../utils/auth';
+
+import { PublishProfileDialog } from './create-publish-profile/PublishProfileDialog';
+import { actionButton } from './styles';
+import { useBoolean } from '@uifabric/react-hooks';
+import Dialog, { DialogFooter } from 'office-ui-fabric-react/lib/Dialog';
+import { styles } from '../design/styles';
+
+// -------------------- CreatePublishProfileDialog -------------------- //
+
+type CreatePublishProfileDialogProps = {
+  projectId: string;
+  scrollToSectionId?: string;
+};
+
+export const CreatePublishProfileDialog: React.FC<CreatePublishProfileDialogProps> = (props) => {
+  const { projectId } = props;
+  const { publishTargets } = useRecoilValue(settingsState(projectId));
+  const { getPublishTargetTypes, setPublishTargets } = useRecoilValue(dispatcherState);
+  const publishTypes = useRecoilValue(publishTypesState(projectId));
+
+  const [dialogHidden, setDialogHidden] = useState(true);
+  const [showAuthDialog, setShowAuthDialog] = useState(false);
+  const [hideDialog, { toggle: toggleHideDialog }] = useBoolean(false);
+
+  const dialogTitle = {
+    title: formatMessage('Create a publish profile to continue'),
+    subText: formatMessage(
+      'To make your bot available as a remote skill you will need to provision Azure resources . This process may take a few minutes depending on the resources you select.'
+    ),
+  };
+  const [current, setCurrent] = useState<{ index: number; item: PublishTarget } | null>(null);
```

Comment:
More descriptive name than current and setCurrent?

Reply:
replace to `currentPublishProfile`